examples/elementstate.js

```diff
@@ -33,34 +33,37 @@ export default async function() {
   `);
 
   // Check state
-  let isVisible = await page.$('.visible').then(e => e.isVisible());
-  let isHidden = await page.$('.hidden').then(e => e.isHidden());
-  let isEditable = await page.$('.editable').then(e => e.isEditable());
-  let isEnabled = await page.$('.enabled').then(e => e.isEnabled());
-  let isDisabled = await page.$('.disabled').then(e => e.isDisabled());
-  let isChecked = await page.$('.checked').then(e => e.isChecked());
-  let isUnchecked = !await page.$('.unchecked').then(e => e.isChecked());
-
-  check(page, {
-    'visible': isVisible,
-    'hidden': isHidden,
-    'editable': isEditable,
-    'enabled': isEnabled,
-    'disabled': isDisabled,
-    'checked': isChecked,
-    'unchecked': isUnchecked,
+  await check(page, {
+    'is visible':
+      async p => p.$('.visible').then(e => e.isVisible()),
+    'is hidden':
+      async p => p.$('.hidden').then(e => e.isHidden()),
+    'is editable':
+      async p => p.$('.editable').then(e => e.isEditable()),
+    'is enabled':
+      async p => p.$('.enabled').then(e => e.isEnabled()),
+    'is disabled':
+      async p => p.$('.disabled').then(e => e.isDisabled()),
+    'is checked':
+      async p => p.$('.checked').then(e => e.isChecked()),
+    'is unchecked':
+      async p => p.$('.unchecked')
+        .then(async e => await e.isChecked() === false),
   });
 
   // Change state and check again
-  await page.$(".unchecked").then(e => e.setChecked(true))
-  await page.$(".checked").then(e => e.setChecked(false))
-
-  let isUncheckedChecked = await page.$(".unchecked").then((e) => e.isChecked());
-  let isCheckedUnchecked = !await page.$(".checked").then((e) => e.isChecked());
-
-  check(page, {
-    isUncheckedChecked: isUncheckedChecked,
-    isCheckedUnchecked: isCheckedUnchecked,
+  await check(page, {
+    'is unchecked checked':
+      async p => p.$(".unchecked")
+        .then(e => e.setChecked(true))
+        .then(() => p.$(".unchecked"))
```

Comment:
> Although having an async check allows us to do this, i don't think we should be doing state changes to the DOM from within the check.

I've tried that approach, but it looked more complicated. It creates a lot of one-usage variables/constants. I'm not against making state changes in a `check` and then checking them if it looks simpler. Especially in a corner case, like `setChecked`. Otherwise, I'd be against it, too.